ViewControllers/Search/SearchTableViewController.swift

```diff
@@ -78,12 +83,19 @@ extension SearchViewController: SearchViewControllerProtocol {
 extension SearchViewController: UISearchResultsUpdating {
     
     func updateSearchResults(for searchController: UISearchController) {
-        
+        var timer: Timer?
+        timer?.invalidate()
         let searchBar = searchController.searchBar
         
         guard let text = searchBar.text, text != "" else {
             return
         }
-        self.logicController.getArtists(name: text)
+        
+        if text.count >= 3 {
+            timer = Timer.scheduledTimer(withTimeInterval: 0.5, repeats: false) { timer in
```

Comment:
1. newline please 
2. close variable timer is not used
3. seems like old timer will not be cancelled...